### PR TITLE
Firestore Database Integration Fixes

### LINKED ISSUE #61: [Feature 0B-2] Firebase Firestore Database Access
Add functionalities for reading from and writing data to the Firebase Firestore Database.

### PR DESCRIPTION
**Pull request by rsrinivasan1**
Updates ```MatchInterator``` to update the Firestore database upon match as well as the realtime. This is required to update the contact list upon match. Also added call to ```FirestoreDbWriter.uploadUser``` when creating a user with romantic type in the ```setRomanticInfo``` method, where it was missing before.

Fixes #61 

Successful match chat image

# Checklist

Please do a self-check to make sure you have satisfied all of the following before marking your PR
as ready for review (place an `x` (the letter x) between `[ ]` to mark complete).

- [x] My PR does not contain unrelated files that may cause conflicts
- [x] I have performed a self-review of my own code
- [x] I have commented my code with javadoc and perhaps in-line comments (particularly in
  hard-to-understand areas)
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have not violated clean architecture principles
- [x] I have opted to use the most general form of datastructures (example: use `List` instead of
  `ArrayList` wherever possible)

### CODE REVIEW

**Review by swaner2 — APPROVED** [review]
Looks good!

File: app/src/main/java/com/group80/uoftinder/MatchInteractor.java [hunk 1]
```diff
@@ -41,8 +42,10 @@ public static boolean checkForMatchAndCreate(User currentUser, User user2) {
             UserRealtimeDbFacade.uploadUser(user2); // re-upload users to the database
             UserRealtimeDbFacade.uploadUser(currentUser);
 
-            sendIntroMessage(currentUser.getUid(),
-                    user2.getUid(), "Hey! We matched with each other!"); // send a message from currentUser to user2
```
Comment (MilenaMijuskovic) [inline app/src/main/java/com/group80/uoftinder/MatchInteractor.java:45]: hehhehreherhheeh 
image

**Review by MilenaMijuskovic — APPROVED** [review]
Nice!

**Review by LanceLi1416 — APPROVED** [review]
Looks good! Happy to see the matching working!

**Pull request merged**